src/containers/classrooms/ClassroomsManagerContainer.jsx

```diff
@@ -6,84 +6,54 @@ import ClassroomsManager from '../../components/classrooms/ClassroomsManager';
 import {
   CLASSROOMS_INITIAL_STATE, CLASSROOMS_PROPTYPES
 } from '../../ducks/classrooms';
+import {
+  PROGRAMS_INITIAL_STATE, PROGRAMS_PROPTYPES
+} from '../../ducks/programs';
 
 export class ClassroomsManagerContainer extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      classroomToDelete: null,
-      showConfirmationDialog: false
-    };
-
-    this.closeConfirmationDialog = this.closeConfirmationDialog.bind(this);
-    this.deleteClassroom = this.deleteClassroom.bind(this);
-    this.maybeDeleteClassroom = this.maybeDeleteClassroom.bind(this);
+  componentDidMount() {
+    if (this.props.selectedProgram) Actions.getClassroomsAndAssignments(this.props.selectedProgram);
   }
 
-  componentDidMount() {
-    Actions.getClassroomsAndAssignments();
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.selectedProgram && this.props.selectedProgram !== nextProps.selectedProgram) {
+      Actions.getClassroomsAndAssignments(nextProps.selectedProgram);
+    }
   }
 
   componentWillUnmount() {
     Actions.classrooms.setClassrooms(CLASSROOMS_INITIAL_STATE.classrooms);
   }
 
-  maybeDeleteClassroom(id) {
-    this.setState({ classroomToDelete: id, showConfirmationDialog: true });
-  }
-
-  closeConfirmationDialog() {
-    this.setState({ classroomToDelete: null, showConfirmationDialog: false });
-  }
-
-  deleteClassroom() {
-    if (this.state.classroomToDelete === null) return;
-
-    Actions.deleteClassroom(this.state.classroomToDelete).then((response) => {
-      // TODO: For API optimization, do we want to instead manually remove the classroom
-      // out of local app state instead of making another API call
-      Actions.getClassroomsAndAssignments();
-      this.closeConfirmationDialog();
-
-      if (response) {
-        Actions.classrooms.setToastState({ status: 'ok', message: 'Classroom deleted' });
-      }
-    });
-  }
-
   render() {
     return (
       <ClassroomsManager
         classrooms={this.props.classrooms}
         classroomInstructions={this.props.classroomInstructions}
         classroomsStatus={this.props.classroomsStatus}
-        classroomToDelete={this.state.classroomToDelete}
-        closeConfirmationDialog={this.closeConfirmationDialog}
-        deleteClassroom={this.deleteClassroom}
```

Comment:
Nyerp this one's bothering me, `deleteClassroom` is removed _here,_ but I'm assuming it should have been added back/moved to _somewhere else._ I still can't tell where it's supposed to go, though.

Reply:
That's right. I shuffled the classroom delete methods into `ClassroomsTableContainer`

Reply:
It was my attempt at making sense of component organization and what should be responsible for what. Clearly, it's all still hard to follow 🤷‍♀️